### PR TITLE
 Stop evaluation during cool down period 

### PR DESCRIPTION
**Pull request by cdlliuy**
#156428400

### CODE REVIEW

File: src/autoscaler/eventgenerator/generator/appEvaluationManager_test.go [hunk 1]
```diff
@@ -100,43 +95,86 @@ var _ = Describe("AppEvaluationManager", func() {
...
+				BeforeEach(func() {
+					getPolicies = func() map[string]*models.AppPolicy {
+						return map[string]*models.AppPolicy{
+							testAppId2: appPolicy2,
+						}
+					}
+				})
+
+				JustBeforeEach(func() {
+					manager.SetCoolDownExpired(testAppId2, fakeTime.Add(time.Duration(30)*time.Second).UnixNano())
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/generator/appEvaluationManager_test.go]: time.Duration(30* testEvaluateInterval)*time.Second ?

File: src/autoscaler/eventgenerator/generator/appEvaluationManager_test.go [hunk 2]
```diff
@@ -100,43 +95,86 @@ var _ = Describe("AppEvaluationManager", func() {
...
+				})
+
+				It("should add triggers to evaluate", func() {
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/generator/appEvaluationManager_test.go]: should add triggers to evaluate after cooldown expired

File: src/autoscaler/eventgenerator/generator/appEvaluationManager.go [hunk 3]
```diff
@@ -125,3 +133,9 @@ func (a *AppEvaluationManager) GetBreaker(appID string) *circuit.Breaker {
 	defer a.lock.Unlock()
 	return a.breakers[appID]
 }
+
+func (a *AppEvaluationManager) SetCoolDownExpired(appID string, expiredAt int64) {
+	a.lock.Lock()
```
Comment (qibobo) [inline src/autoscaler/eventgenerator/generator/appEvaluationManager.go]: Is it better to provide two locks to setcooldown and circuitbreaker as they are independent?

**Comment by cdlliuy** [comment]
@boyang9527  @qibobo , please see the update

**Comment by qibobo** [comment]
LGTM

**Comment by boyang9527** [comment]
LGTM

**Pull request merged**